src/charm.py

```diff
@@ -1594,6 +1604,18 @@ def get_available_resources(self) -> Tuple[int, int]:
 
         return cpu_cores, allocable_memory
 
+    def on_deployed_without_trust(self) -> None:
+        """Blocks the application and returns a specific error message for deployments made without --trust."""
+        self.unit.status = BlockedStatus(
+            "Unauthorized access to k8s resources. Is the app trusted? See logs"
+        )
```

Comment:
Please consider to change Juju status to mimic `nginx-ingress-integrator`: 

```
nginx-ingress-integrator/0*  blocked      idle       10.1.204.237         Insufficient permissions, try: `juju trust <nginx-ingress-integrator> --scope=cluster`
```

P.S. IMHO, you know the app name, so you can print the fix command copy&paste'able:
```
juju deploy postgresql-k8s mypostgresql-k8s
...
...  Insufficient permissions, try: `juju trust mypostgresql-k8s --scope=cluster`
```